### PR TITLE
Fix faster search results for dropdown

### LINKED ISSUE #217: (eng spike) faster search results
most people we've observed using the site, type in the school name in the search bar rather than finding it on the map or list view. the current search results sometimes take a while to appear (especially the first time typing). for example, start typing "washington" and it would be nice if washington appeared sooner. it would also be nice if typing "oconnell" turned up "o'connell" even though there's an apostrophe.

it probably loads faster after some caching, i don't know the technical details right now.

### PR DESCRIPTION
**Pull request by mattgianni**
Fixes #217. Currently the map component issues a fetch and a DB query with every keystroke during the search. This is leading to the choppy experience. This PR replaces the fetch and query with an in-memory filter.

### CODE REVIEW

File: src/pages/map.tsx [hunk 1]
```diff
@@ -75,15 +75,17 @@ const Map: React.FC<Props> = (props) => {
...
-    const response = await fetch(
-      `/api/searchSchools?searchTerm=${searchTerm}`,
-    ).then((res) => res.json() as Promise<{ schools: School[] }>);
-
-    return response.schools.map((school) => ({
-      label: school.name,
-      value: school.name,
-      item: school,
-    }));
+    const results = props.schools
```
Comment (thomhickey) [inline src/pages/map.tsx:78]: Fantastic, @mattgianni  ... I wondered why we didn't just .filter.map, but hadn't looked into it enough to realize that props.schools was already in component state! Great fix.

**Comment by thomhickey** [comment]
approved

File: src/pages/map.tsx (hunk 1 above)
Comment (thomhickey) [reply src/pages/map.tsx:78]: Also avoids possible problems with hydration where the db was updated, but app not rebuilt, causing api to return schools that weren't in the hydrated list. Great fix.

File: src/pages/map.tsx [hunk 2]
```diff
@@ -75,15 +75,17 @@ const Map: React.FC<Props> = (props) => {
...
+      .filter(({ name }) =>
+        name.toUpperCase().includes(searchTerm.toUpperCase()),
+      )
+      .map((school) => ({
+        label: school.name,
+        value: school.name,
+        item: school,
+      }));
+
+    return results;
```
Comment (thomhickey) [inline src/pages/map.tsx:88]: maybe just return the .filter.map? no need for the results const.

File: src/pages/map.tsx (hunk 1 above)
Comment (mattgianni) [reply src/pages/map.tsx:78]: thx, ... i thought it was going to be a little tougher too, but the schools were already being pulled into the component.

File: src/pages/map.tsx (hunk 2 above)
Comment (mattgianni) [reply src/pages/map.tsx:88]: works for me!

**Comment by nickvisut** [comment]
Nice catch! Makes sense to not be making another call and furthermore keeping in-memory since slim chance the data goes stale during.

**Pull request merged**